akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpResponseRendererFactory.scala

```diff
@@ -33,22 +33,22 @@ private[http] class HttpResponseRendererFactory(serverHeader: Option[headers.Ser
     }
 
   // as an optimization we cache the Date header of the last second here
-  @volatile private[this] var cachedDateHeader: (Long, Array[Byte]) = (0L, null)
-
+  private[this] var cachedDateHeader: (Long, Array[Byte]) = (0L, null)
   private def dateHeader: Array[Byte] = {
-    var (cachedSeconds, cachedBytes) = cachedDateHeader
-    val now = System.currentTimeMillis
-    if (now - 1000 > cachedSeconds) {
-      cachedSeconds = now / 1000
+    var cachedSeconds = cachedDateHeader._1
+    val now = currentTimeMillis()
+    val nowSeconds = now / 1000
+    if (nowSeconds > cachedSeconds) {
+      cachedSeconds = nowSeconds
       val r = new ByteArrayRendering(48)
-      dateTime(now).renderRfc1123DateTimeString(r ~~ headers.Date) ~~ CrLf
-      cachedBytes = r.get
-      cachedDateHeader = cachedSeconds -> cachedBytes
+      DateTime(now).renderRfc1123DateTimeString(r ~~ headers.Date) ~~ CrLf
+      cachedDateHeader = cachedSeconds -> r.get
     }
-    cachedBytes
+    cachedDateHeader._2
```

Comment:
Yes. So this makes it possible that a server produces two requests A and B going out over the same connection to the same client in that sequence, where A has a `Date` header with a later timestamp than B.
I'm not sure whether this has the potential of breaking anything.

Having thought about it a bit more I'd probably vote for being on the safe side and **keeping the volatile tuple** rather than opening up a hole for a potentially extremely hard to debug problem in complicated enterprise deployments (which proxies, caches, and what-not).